bitcoin/BRPeerManager.c

```diff
@@ -1679,7 +1673,7 @@ void BRPeerManagerDisconnect(BRPeerManager *manager)
         p = manager->connectedPeers[i - 1];
         manager->connectFailureCount = MAX_CONNECT_FAILURES; // prevent futher automatic reconnect attempts
         BRPeerDisconnect(p);
-        if (BRPeerConnectStatus(p) == BRPeerStatusConnecting) manager->peerThreadCount--; // waiting for network
+        while (BRPeerConnectStatus(p) == BRPeerStatusConnecting) BRPeerDisconnect(p);
```

Comment:
This makes me leery (endless loop anyone?) but the logic *should* be that this only happens when the thread is being spawned. Once spawned, the BRPeer thread will set the status to connected or disconnected. In either case, this will break out (*fingers crossed but also seen in testing*).